src/rewards/ca/mod.rs

```diff
@@ -0,0 +1,201 @@
+mod funding;
+mod lottery;
+
+use crate::community_advisors::models::{AdvisorReviewRow, ReviewScore};
+use lottery::TicketsDistribution;
+use std::cmp::Ordering;
+
+use std::collections::{HashMap, HashSet};
+
+pub use crate::rewards::ca::funding::ProposalRewardSlots;
+pub use funding::{FundSetting, Funds};
+
+pub type Ca = String;
+pub type ProposalId = String;
+// Lets match to the same type as the funds, but naming it funds would be confusing
+pub type Rewards = Funds;
+
+type ProposalsFunds = HashMap<ProposalId, ProposalReward>;
+pub type CaRewards = HashMap<Ca, Rewards>;
+pub type ProposalsReviews = HashMap<ProposalId, Vec<AdvisorReviewRow>>;
+pub type ApprovedProposals = HashSet<ProposalId>;
+
+enum ProposalFundsState {
+    // Proposal has the exact quantity reviews to be rewarded
+    Exact,
+    // Proposal has less reviews as needed so some of the funds should go back into the rewards pool
+    Unfilled(Funds),
+    // It has more reviews than fitted rewards
+    OverLoaded,
+}
+
+struct ProposalReward {
+    pub state: ProposalFundsState,
+    pub funds: Funds,
```

Comment:
Is it necessary to store funds per proposal? Apart from bonuses, which can be stored here, each review is worth the same irregardless of the proposal